gossip/src/cluster_info.rs

```diff
@@ -213,7 +213,7 @@ pub struct ClusterInfo {
     /// The network
     pub gossip: RwLock<CrdsGossip>,
     /// set the keypair that will be used to sign crds values generated. It is unset only in tests.
-    pub keypair: Arc<Keypair>,
+    keypair: RwLock<Arc<Keypair>>,
```

Comment:
A new RwLock here so keypair can be updated. cc: @behzadnouri

Reply:
Why are these `RwLock<Arc<...>>` as opposed to `Arc<RwLock<...>>`?

Reply:
Heh, good question:
1. One can't clone/copy a `Keypair`, so we need an `Arc<Keypair>` to be able to generally pass the keypair around.  Shoving a `Rwlock` between the `Arc` and `Keypair` would cause a ton of code churn and `read().unlock()` overhead on each access
2. There's no need for an `Arc` around the `RwLock` since everybody holds an `Arc` to `ClusterInfo` already